### ISSUE TITLE
Vacuum stops cleaning - add option to enable logging?

### ISSUE THREAD

**Issue by Hololinked** [homebridge-plugins/homebridge-sharkiq#15]
**Is your feature request related to a problem? Please describe:**

Sometimes the vaccuum only runs a couple of minutes before returning to the dock. I do not know what causes this.

**Describe the solution you'd like:**

I think first thing that is needed is the option to toggle logging with a way to identify what caused the vaccuum to stop the cleaning session; i.e, was it's own decision (shark issue); was it due to the plugin; or was it due to Apple home (apple issue).

**Describe alternatives you've considered:**

**Additional context:**

It really seems to be hit or miss whether the vacuum will finish a cleaning session or stop early. The vaccuum is perfectly maintained by the way.

**Comment by Bubba8291** [comment]
I was considering adding the option to have an optional sensor that is on when the vacuum is in an error state. With the nature of HomeKit, it's hard to display specific errors. So when I add this, it would be a sensor object that's on when there's an error.

Something I could add though is a trigger to run a script or webhook when the vacuum is in an error state, which could include the specific error.

**Comment by Hololinked** [comment]
Yeah, I'm not sure! Have you experienced the issue that I have described? I've been using this plug-in ever since you released it  and that is the only issue that I've had.

If you're referring to errors as reported by the vacuum unit itself (dust cup full, etc), I'm not sure that it would help in this case because it doesn't tell me of any errors. It simply stops and returns to the dock as though it's finished the session or told to stop cleaning.

It happened to me twice in a row just now when I asked Siri to run it, with it returning itself to the dock about five minutes (it takes around 25 minutes to run a full session). I then tried a third time by activating it manually in the Home app, and it succeeded.

Regardless, just want to take this opportunity to thank you for this plug-in. It is an absolute godsend and 1000 times more stable than the Home Assistant plug-in I was using before.

**Comment by Bubba8291** [comment]
@Hololinked From the most recent updates with the plugin, have the enhancements you mentioned been improved?

**Comment by bwp91** [comment]
The plugin now has debug logging (enable it in the plugin settings) which will show state changes and any API messages around when the vacuum returns to the dock — that should help tell apart a Shark-side decision, a plugin issue, or a HomeKit one. If you can capture a session where it stops early with debug on and share the log, I'm happy to look. Otherwise this is largely a Shark-firmware behaviour rather than something the plugin controls.
